crates/core/src/network/net_util.rs

```diff
@@ -30,36 +30,47 @@ use restate_types::net::{AdvertisedAddress, BindAddress};
 
 use crate::{cancellation_watcher, ShutdownError, TaskCenter, TaskKind};
 
-pub fn create_tonic_channel_from_advertised_address<T: CommonClientConnectionOptions>(
+pub fn create_tonic_channel<T: CommonClientConnectionOptions>(
```

Comment:
An approach where we accept a closure to customize the endpoint might be a nicer alternative if `CommonClientConnectionOptions` gets any more complex, but I'm punting on that for now.